test/endpoints/baz/baz_simpleservice_method_ping_test.go

```diff
@@ -97,3 +113,109 @@ func BenchmarkPing(b *testing.B) {
 	b.StopTimer()
 	gateway.Close()
 }
+
+func TestPing(t *testing.T) {
+	gateway, err := testGateway.CreateGateway(t, testConfig, testOptions)
+	if !assert.NoError(t, err, "got bootstrap err") {
+		return
+	}
+	defer gateway.Close()
+
+	callCounter := 0
+
+	fakePing := func(
+		ctx context.Context,
+		reqHeaders map[string]string,
+	) (*clientsBazBase.BazResponse, map[string]string, error) {
+		callCounter++
+
+		return &clientsBazBase.BazResponse{
+			Message: "a message",
+		}, nil, nil
+	}
+
+	gateway.TChannelBackends()["baz"].Register(
+		"SimpleService",
+		"ping",
+		bazClient.NewSimpleServicePingHandler(fakePing),
+	)
+
+	res, err := gateway.MakeRequest("GET", "/baz/ping", nil, nil)
+
+	if !assert.NoError(t, err, "got request error") {
+		return
+	}
+
+	assert.Equal(t, 1, callCounter)
+	assert.Equal(t, 200, res.StatusCode)
+
+	bytes, err := ioutil.ReadAll(res.Body)
+	if !assert.NoError(t, err, "got read error") {
```

Comment:
we expect `NoError()` but if there is an error then we return; we cannot expect `Error()` because then the test would not pass.